File: Cpp/Product.cpp
#include<iostream>
#include<string>
using namespace std;

// Base class: Product
class Product {
private:
    int id;         // Unique product ID
    string name;    // Name of the product
    string brand;   // Brand of the product
    int price;      // Price of the product

public:
    // Default constructor (initializes attributes with default values)
    Product() {
        this->id = 0;
        this->name = "";
        this->brand = "";
        this->price = 0;
    }

    // Parameterized constructor (allows initialization with specific values)
    Product(int id, string name, string brand, int price) {
        this->id = id;
        this->name = name;
        this->brand = brand;
        this->price = price;
    }

    // Getter methods (to access private attributes)
    int getId() {
        return this->id;
    }

    string getName() {
        return this->name;
    }

    string getBrand() {
        return this->brand;
    }

    int getPrice() {
        return this->price;
    }

    // Setter methods (to modify private attributes)
    void setName(string name) {
        this->name = name;
    }

    void setBrand(string brand) {
        this->brand = brand;
    }

    void setPrice(int price) {
        this->price = price;
    }

    // Destructor
    ~Product() {
    };
};

File: PHP/Product.php
<?php
class Product {
    // Properties dengan tipe data ketat (strict typing)
    private int $id;       // ID unik produk
    private string $name;  // Nama produk
    private string $brand; // Merek produk
    private int $price;    // Harga produk
    private string $image;

    // Constructor default (nilai awal jika tidak diberikan parameter)
    public function __construct(int $id = 0, string $name = "", string $brand = "", int $price = 0, string $image = "") {
        $this->id = $id;
        $this->name = $name;
        $this->brand = $brand;
        $this->price = $price;
        $this->image = $image;
    }

    // Getter (mengambil nilai properti)

    // Ambil ID produk
    public function getId(): int {
        return $this->id;
    }

    // Ambil nama produk
    public function getName(): string {
        return $this->name;
    }

    // Ambil brand produk
    public function getBrand(): string {
        return $this->brand;
    }

    // Ambil harga produk
    public function getPrice(): int {
        return $this->price;
    }

    public function getImage(): string { 
        return $this->image; 
    }

    // Setter (mengubah nilai properti)

    // Ubah nama produk
    public function setName(string $name): void {
        $this->name = $name;
    }

    // Ubah brand produk
    public function setBrand(string $brand): void {
        $this->brand = $brand;
    }

    // Ubah harga produk
    public function setPrice(int $price): void {
        $this->price = $price;
    }

    public function setImage(string $image): void { 
        $this->image = $image; 
    }
}
